### PR TITLE
[release] Disable NFS for many nodes release test

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

NFS is disabled for all release tests due to this many nodes test, since 2k nodes is too many clients for Google Filestore. This PR opts out of NFS for this test so that NFS can be enabled for all other tests.

## Related issue number
Required for https://github.com/ray-project/ray/pull/37142

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jjyao — APPROVED** [review]

**Comment by justinvyu** [comment]
Release tests kicked off here: https://buildkite.com/ray-project/release-tests-pr/builds/45074#01894234-0163-43a3-8309-97b52b6aa35a

Waiting for passing before merge.

**Comment by justinvyu** [comment]
Both tests launch successfully (no NFS errors) but are timing out: https://buildkite.com/ray-project/release-tests-pr/builds/45239#0189474c-9bc5-4cdd-9947-ff500cdec990

Probably due to an unrelated issue from a recent core commit -- @jjyao do you think this is ok to merge, as it's fixing the NFS issue, and no Ray code has been touched?

**Pull request merged**
